app/modules/intelligence/agents/custom_agents/custom_agent_router.py

```
@@ -0,0 +1,119 @@
+            tasks=[task.model_dump(exclude_unset=True) for task in request.tasks],
+        )
+    except HTTPException as he:
+        raise he
+    except Exception as e:
+        logger.error(f"Error creating custom agent: {str(e)}")
+        raise HTTPException(status_code=500, detail="Internal server error")
+
+@router.delete("/{agent_id}", response_model=dict)
+async def delete_custom_agent(
+    agent_id: str,
+    db: Session = Depends(get_db),
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)